subprojects/core/src/main/java/org/gradle/api/internal/file/AbstractFileCollection.java

```diff
@@ -155,27 +157,29 @@ public FileCollection stopExecutionIfEmpty() {
         return this;
     }
 
+    @Deprecated
+    @Override
     public Object asType(Class<?> type) throws UnsupportedOperationException {
-        if (type.isAssignableFrom(Set.class)) {
-            return getFiles();
-        }
-        if (type.isAssignableFrom(List.class)) {
-            return new ArrayList<File>(getFiles());
+        if (type.isAssignableFrom(Object[].class)) {
```

Comment:
Why the new `Object[]` cast?

Reply:
This is to prevent a deprecation warning being printed for the `Object[]` case.